src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class NullableWalker
+    {
+#if DEBUG
+        private sealed class DebugVerifier : BoundTreeWalker
```

Comment:
>DebugVerifier [](start = 29, length = 13)

General comment: I didn't understand why this type needs to special-case so many `Visit` methods. I would have assumed that we'd just visit all children by default (and skip verification for certain node kinds).

For example, I don't understand why we need to override `VisitDelegateCreationExpression`. The default visit handles `null` receivers (as no-op) and if the argument is a `BoundMethodGroup` (which is a `BoundExpression`) that would be handled by `Visit(BoundNode)` which would call `VerifyNode`.

Reply:
It needs so many special case visits because it needs to deal with things that the NullableWalker does or does not do. `VisitDelegateCreationExpression`, for example: there _is_ a node as a receiver for a static method sometimes (in error scenarios, mainly). `NullableWalker` never visits these nodes, so it's not as simple as just letting the default logic handle it.

---
In reply to: [265383260](https://github.com/dotnet/roslyn/pull/34017#discussion_r265383260) [](ancestors = 265383260)